plugins/org.eclipse.xtext/src/org/eclipse/xtext/generator/URIBasedFileSystemAccess.xtend

```diff
@@ -73,9 +77,10 @@ class URIBasedFileSystemAccess extends AbstractFileSystemAccess2 {
 	
 	override generateFile(String fileName, String outputCfgName, CharSequence contents) {
 		val uri = getURI(fileName, outputCfgName)
-		generateTrace(fileName, outputCfgName, contents)
 		val encoding = getEncoding(uri)
```

Comment:
Here we need to take encoding from the language of the generated file rather than the language of the injector that instantiated the class. Right now it seems like all files will be produced with the runtime encoding.